ferminet/train.py

```diff
@@ -381,6 +383,12 @@ def train(cfg: ml_collections.ConfigDict):
       steps=cfg.mcmc.steps,
       atoms=atoms_to_mcmc,
       one_electron_moves=cfg.mcmc.one_electron)
+  hmc_step = hmc.make_hmc_step(network,
```

Comment:
Does it make sense to always create the HMC step, and then only decide whether or not to use it later? Seems like it makes sense to only create it if we're going to use it.